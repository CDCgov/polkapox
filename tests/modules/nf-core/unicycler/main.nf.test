nextflow_process {

    name "Test Process UNICYCLER"
    script "modules/nf-core/unicycler/main.nf"
    process "UNICYCLER"

    test("Unicycler Test") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_sub', single_end: false ],
                    [
                        file("${projectDir}/tests/nf-test-data/test_sub_1.fastp.fastq.gz", checkIfExists: true),
                        file("${projectDir}/tests/nf-test-data/test_sub_2.fastp.fastq.gz", checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
