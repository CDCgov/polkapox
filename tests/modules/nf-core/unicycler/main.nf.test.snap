{
    "Unicycler Test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "gfa": [
                    
                ],
                "log": [
                    
                ],
                "scaffolds": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:22:56.90930693"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "gfa": [
                    
                ],
                "log": [
                    
                ],
                "scaffolds": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-07T13:39:47.290721642"
    }
}